redis/client.py

```diff
@@ -2733,17 +2733,31 @@ def xrevrange(self, name, max='+', min='-', count=None):
 
         return self.execute_command('XREVRANGE', name, *pieces)
 
-    def xtrim(self, name, maxlen, approximate=True):
+    def xtrim(self, name, maxlen=None, approximate=True, minid=None,
+              limit=None):
         """
         Trims old messages from a stream.
         name: name of the stream.
         maxlen: truncate old stream messages beyond this size
         approximate: actual stream length may be slightly more than maxlen
+        minin: the minimum id in the stream to query
+        limit: specifies the maximum number of entries to retrieve
         """
-        pieces = [b'MAXLEN']
+        pieces = []
+        if maxlen is not None:
```

Comment:
MAXLEN and MINID are mutually exclusive strategies, should this be an elif?

Reply:
Added exclusion + tests.